cpp/oneapi/dal/algo/kmeans/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -105,7 +107,84 @@ struct infer_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
+                                                 { 0, -1 },
+                                                 sparse_indexing::zero_based,
+                                                 sycl::usm::alloc::device);
+        auto values = pr::ndarray<Float, 1>::wrap(arr_val.get_data(), arr_val.get_count());
+        auto column_indices =
+            pr::ndarray<std::int64_t, 1>::wrap(arr_col.get_data(), arr_col.get_count());
+        auto row_offsets =
+            pr::ndarray<std::int64_t, 1>::wrap(arr_row.get_data(), arr_row.get_count());
+        auto arr_centroid_squares =
+            pr::ndarray<Float, 1>::empty(queue, cluster_count, sycl::usm::alloc::device);
+        auto arr_data_squares =
+            pr::ndarray<Float, 1>::empty(queue, row_count, sycl::usm::alloc::device);
+        auto data_squares_event =
+            compute_data_squares(queue, values, column_indices, row_offsets, arr_data_squares);
+
+        auto distances = pr::ndarray<Float, 2>::empty(queue,
+                                                      { row_count, cluster_count },
+                                                      sycl::usm::alloc::device);
+
+        auto arr_closest_distances =
+            pr::ndarray<Float, 2>::empty(queue, { row_count, 1 }, sycl::usm::alloc::device);
+        auto arr_centroids = pr::table2ndarray<Float>(queue,
+                                                      input.get_model().get_centroids(),
+                                                      sycl::usm::alloc::device);
+        auto arr_responses =
+            pr::ndarray<std::int32_t, 2>::empty(queue, { row_count, 1 }, sycl::usm::alloc::device);
+
+        auto centroid_squares_event = kernels_fp<Float>::compute_squares(queue,
+                                                                         arr_centroids,
+                                                                         arr_centroid_squares,
+                                                                         { data_squares_event });
+        auto assign_event = assign_clusters(queue,
+                                            values,
+                                            column_indices,
+                                            row_offsets,
+                                            arr_data_squares,
+                                            arr_centroids,
+                                            arr_centroid_squares,
+                                            distances,
+                                            arr_responses,
+                                            arr_closest_distances,
+                                            { data_squares_event, centroid_squares_event });
+        auto objective_function =
+            calc_objective_function(queue, arr_closest_distances, { assign_event });
+        {
+            // Reduce objective function value over all ranks
+            auto obj_func_reduce_event = comm.allreduce(objective_function);
```

Comment:
minor comment, but you can reduce number of rows via `comm.allreduce(objective_function).wait()` and it will be done in the same style as other comm functions